### ISSUE TITLE
HMR don't detect change made with fs.writeFile from nodejs

### ISSUE THREAD

**Issue by apiel** [webpack/webpack#9094]
# Bug report

**What is the current behavior?**

When a nodejs script is writing code in the "src" folder watched by HMR, the change detection are unstable. Sometime it get detected, sometime not.

**If the current behavior is a bug, please provide the steps to reproduce.**

```
import { writeFileSync } from 'fs';

writeFileSync('src/data.js', `export const value = ${Math.random()};`);
```
Run this script several time, at some point HMR should not detect the change.

(I am using create-react-app)

**What is the expected behavior?**

As soon as I am writing a file in src folder, HMR should detect it and reload the page.

**Other relevant information:**
webpack version:  from the latest version of create-react-app
Node.js version: v11.10.1
Operating System: ubuntu
Additional tools:

**Comment by evilebottnawi** [comment]
Already answered. hmr !== files changes. you need watch files

**Comment by evilebottnawi** [comment]
Also be ensure no cached enabled, some loader/plugins or tools can cache you files

**Comment by apiel** [comment]
I don't really understand your answer, I am sorry. 
I know nothing about webpack and even less about HMR. As a noob, I beleive that when I save a file from my IDE (vscode), it is writing it in my file system. So I try to reproduce it with `fs.writeFile` from node.js that seem to be the same as writing file in file system. Therefor, since `fs.writeFile` is not detected by HMR, I see it as a bug.
Of course, I beleive you when you say me it doesn't work like this, but I don't understand your answer `hmr !== files`. You asked me to use the template, I did. But your answer are absolutely not clear. Could you give more details, maybe some links...

I can give you the code where I have the issue:
https://github.com/apiel/isomor/blob/master/packages/isomor-transpiler/lib/index.ts#L113
but it's too much code and I doubt that you want to digg into it...

If you want a real test environment:

```bash
npx isomor-react-app my-app
cd my-app
yarn dev
```

then go in `my-app/src-isomor` and edit some file. At some point it will not get reloaded.

I would really appreciate some help. (by the way, I also opened an issue in create-reat-app and stackoverflow)

**Comment by evilebottnawi** [comment]
What is your os? Try to use `usePolling` option for `chokidar` (https://github.com/paulmillr/chokidar#api)

**Comment by apiel** [comment]
![image](https://user-images.githubusercontent.com/5873198/57229773-2f33e680-7017-11e9-926a-2995ed521573.png)

The problem is not coming from `chokidar` used in my transpiler. The transpiler work very well. The problem is when the transpiler the write file to the `src` folder webpack don't detect the change (or do you mean webpack use `chokidar` and I should try to use CHOKIDAR_USEPOLLING=1 when I start my react application?).

**Comment by evilebottnawi** [comment]
> do you mean webpack use chokidar and I should try to use CHOKIDAR_USEPOLLING=1 when I start my react application

Yes, some filesystem version have bugs with watch, also try to use `10` node, node also have some bugs in watch, we need to determine what problem is not environment side

**Comment by apiel** [comment]
Ok, thanks I will give a try.
I still would love to understand, what is the difference between VsCode writing a file and nodejs `fs.writeFile`? Why does HMR behave differently?

**Comment by evilebottnawi** [comment]
HMR is not file reloading, HMR - it is code inside you files what allow to you reload page without full reload, using `fs.writeFile` just create `fs` event. So if your code has `hmr` code you will get hot reloading (not full page) otherwise you page was reloaded. So you problem not related to HMR, it is problem with `fs` event
